File: ios/Classes/HardwareStreamHandler.swift
import Flutter
import UIKit

/**
 * Unified Stream Hub for all iOS hardware subsystems.
 * Directs global event delivery and manages native lifecycle connections.
 */
public class HardwareStreamHandler: NSObject, FlutterStreamHandler {
    private var eventSink: FlutterEventSink?
    
    // Subsystems
    private let camera: HardwareCameraManager
    private let bluetooth: HardwareBluetoothManager
    private let location: HardwareLocationManager
    private let sensor: HardwareSensorManager
    
    public init(camera: HardwareCameraManager, bluetooth: HardwareBluetoothManager, location: HardwareLocationManager, sensor: HardwareSensorManager) {
        self.camera = camera
        self.bluetooth = bluetooth
        self.location = location
        self.sensor = sensor
    }
    
    public func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        self.eventSink = events
        camera.setEventSink(events)
        bluetooth.setEventSink(events)
        location.setEventSink(events)
        sensor.setEventSink(events)
        return nil
    }
    
    public func onCancel(withArguments arguments: Any?) -> FlutterError? {
        self.eventSink = nil
        camera.setEventSink(nil)
        bluetooth.setEventSink(nil)
        location.setEventSink(nil)
        sensor.setEventSink(nil)
        return nil
    }
}

File: android/src/main/kotlin/com/example/nexora_sdk/HardwareStreamHandler.kt
package com.example.nexora_sdk

import android.content.Context
import io.flutter.plugin.common.EventChannel

/**
 * Unified Stream Hub for all hardware subsystems.
 * Distributes the event sink to managers and manages lifecycle.
 */
class HardwareStreamHandler(
    private val context: Context,
    private val sensorManager: HardwareSensorManager,
    private val cameraManager: HardwareCameraManager,
    private val bluetoothManager: HardwareBluetoothManager,
    private val locationManager: HardwareLocationManager
) : EventChannel.StreamHandler {

    private var eventSink: EventChannel.EventSink? = null

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        eventSink = events
        
        // Connect all managers to the unified sink
        sensorManager.setEventSink(events)
        cameraManager.setEventSink(events)
        bluetoothManager.setEventSink(events)
        locationManager.setEventSink(events)
    }

    override fun onCancel(arguments: Any?) {
        eventSink = null
        sensorManager.setEventSink(null)
        cameraManager.setEventSink(null)
        bluetoothManager.setEventSink(null)
        locationManager.setEventSink(null)
    }
}
